File: kotlin-multiplatform/src/commonMain/kotlin/com/ricoh360/thetableclient/service/data/values/ShootingFunction.kt
package com.ricoh360.thetableclient.service.data.values

/**
 * Shooting function.
 */
enum class ShootingFunction : SerialNameEnum {
    /**
     * Undefined value
     */
    UNKNOWN,

    /**
     * normal
     */
    NORMAL {
        override val serialName: String = "normal"
    },

    /**
     * Self timer
     *
     * RICOH THETA X is not supported.
     */
    SELF_TIMER {
        override val serialName: String = "selfTimer"
    },

    /**
     * My setting
     */
    MY_SETTING {
        override val serialName: String = "mySetting"
    },
    ;

    companion object {
        val keyName: String
            get() = "shootingFunction"

        internal fun get(serialName: String?): ShootingFunction? {
            return SerialNameEnum.get(serialName, values(), UNKNOWN)
        }
    }
}

File: react-native/src/service/values/shooting-function.ts
/**
 * Shooting function.
 */
export const ShootingFunctionEnum = {
  /**
   * Undefined item
   */
  UNKNOWN: 'UNKNOWN',
  /** Normal shooting function */
  NORMAL: 'NORMAL',
  /** Self-timer shooting function(RICOH THETA X is not supported.) */
  SELF_TIMER: 'SELF_TIMER',
  /** My setting shooting function */
  MY_SETTING: 'MY_SETTING',
} as const;

/** type definition of ShootingFunctionEnum */
export type ShootingFunctionEnum =
  (typeof ShootingFunctionEnum)[keyof typeof ShootingFunctionEnum];
